### ISSUE TITLE
User is able to get a cms-page into a broken state, unrecoverable through UI.

### ISSUE THREAD

**Issue by derek-adair** [divio/django-cms#6025]
### Summary
*Very easily replicated by adding a broken plugin to a page.  "Broken" as in a typo in the template path.*  This is a minor pain in my development workflow as I just delete the entire page, fix the bug, and re-create the page.  However, this could have serious repercussions if a user is forced to either contact me to fix it on the database level, or delete the whole page and start over.  

This is one of those grey area feature request/bugs/design decisions.  Apologies if this is not the right forum for such a submission. I believe this is mostly only an issue with development and/or broken plugins in general, but _there may be ways to achieve this state in a stable environment with configuration errors or user oversight._  I've not used Django-cms in prod yet but I can imagine this being quite a headache if a user manages to screw something up (I always assume they will).

It's entirely possible i'm missing something in the admin UI for me to be able to direct users to undo a page edit, or maybe even manage page structure without having to delte

### Expected behaviour
When a user breaks the page structure in the UI, it should be recoverable. The view where a user is able to edit page structure seems to be tied to the rendering of the actual page somehow.

* Maybe some sort of "revert changes" thing?
* Maybe separate the editing of the structure from the rendering? I've not dove into how this works, maybe it's not worth.

#### Actual behaviour
 Unless i'm missing something here... I am forced to go into the database and fix this manually. 

#### Environment

* Python version: 2.7 / 3.3
* Django version: 1.8 / 1.11
* django CMS version: 3.4.4

**Comment by derek-adair** [comment]
Is there a way to discard non-published changes?

**Comment by czpython** [comment]
Closing as duplicate of https://github.com/divio/django-cms/issues/3157

A typo in the template path of a page is not something I personally would consider a bug, we could definitely handle it better but in general I think we can address it using the cms check command. The fix is just to update the page objects from old path to new path.

In regards to exception handling of plugins, we've now decoupled the structure rendering from the content rendering, this change will be available in 3.5 and allows you to delete faulty plugins from the structure mode without rendering the page.
